taipy/core/_manager/_manager.py

```diff
@@ -59,6 +59,13 @@ def _get_all(cls) -> List[EntityType]:
         """
         return cls._repository._load_all()
 
+    @classmethod
+    def _get_all_by(cls, by) -> List[EntityType]:
```

Comment:
Signatures for methods like get_all_by should be like : 
`def _get_all_by(cls, attribute_name, attribute_value) -> List[EntityType]:`
otherwise it is really error prone.

Reply:
I'm not in love with the "by" name. Do you have any better idea? 
cc @toan-quach

Reply:
that is quite standard for document-based queries.

Reply:
We are not searching y attribute name or value. We are just searching if something contains this value. 
What do you think  of this signature `def _get_all_by(cls, should_contain: str)`

Reply:
how about def _get_all_if_contains(cls, query/should_contain)?

Reply:
@jrobinAV

Reply:
I personally prefer `get_all_by` which is more generic and does not expose how we are searching for the attribute provided.
We are at the manager level. We absolutely don't want to expose how it is implemented. By the way, at the repository level, we don't want either to expose the fact that we are just checking for a string matching in a file.

Reply:
I would prefer `get_all_by_key` for more clarity or just `find` for simplicity.

@jrobinAV I thought that anyone can read the code so there is nothing we can and should hide. For the end users, normally they would not access the manager and repository directly so I wonder if this is a problem.

Reply:
Yes, indeed we cannot hide it.
What I meant is that the logic of the file system implementation should stay at the repository level. The API (the method name) should be generic so we don't care if it is implemented with FS or database. Basically, we want to separate responsibilities between managers and repositories.